### PR TITLE
Precompiled Headers & Faster CI

### PR DESCRIPTION
**Pull request by Nopey**
small extra change: renamed bktool's function bk_info bk_getinfo because `bk_info` is already the name of the struct bk_info_t typedef

### CODE REVIEW

File: tools/bktool/main.c [hunk 1]
```diff
@@ -882,7 +882,7 @@ int main(int argc, char *argv[]) {
         } else if(keylist->count > 0) {
             bk_keylist();
         } else {
-            bk_info(&bk);
+            bk_getinfo(&bk);
```
Comment (mrannanj) [inline tools/bktool/main.c:885]: This change seems to be maybe part of some other commit?

**Review by mrannanj — CHANGES_REQUESTED** [review]

File: tools/bktool/main.c (hunk 1 above)
Comment (Vagabond) [reply tools/bktool/main.c:885]: it was renamed to not conflict with the struct of the same name.

**Review by Vagabond — APPROVED** [review]
This all looks reasonable to me

File: .github/workflows/compilation.yml [hunk 2]
```diff
@@ -44,18 +44,18 @@ jobs:
       fail-fast: false
       matrix:
         config:
-        - { name: "gcc 14",   cc: gcc-14,   tidy: "On"  }
```
Comment (Vagabond) [inline .github/workflows/compilation.yml:47]: GCC has tidy turned off now?

File: tools/bktool/main.c (hunk 1 above)
Comment (mrannanj) [reply tools/bktool/main.c:885]: I don't object to the change, but it doesn't seem to be in the correct commit.

File: .github/workflows/compilation.yml (hunk 2 above)
Comment (mrannanj) [reply .github/workflows/compilation.yml:47]: Well, I guess this is about clang-tidy, so it would be a mistake to have it on for gcc.

**Review by katajakasa — APPROVED** [review]

File: .github/workflows/compilation.yml (hunk 2 above)
Comment (Nopey) [reply .github/workflows/compilation.yml:47]: I've moved the clang-tidy run from gcc unittests to clang unittests

clang-tidy can only consume the precompiled header files generated by a matching version of clang
Enabling Tidy in a gcc build will disable PCH and emit a warning (I added a check in cmakelists.txt)
This is also why I bumped clang from 16 to 18, to match our clang-tidy version.

File: tools/bktool/main.c (hunk 1 above)
Comment (Nopey) [reply tools/bktool/main.c:884]: Moved it into its own commit :+1:

**Review by mrannanj — APPROVED** [review]

**Pull request merged**